app/Http/Controllers/PluginPageController.php

```diff
@@ -23,7 +24,7 @@ public function __invoke(PluginManagerInterface $manager, Plugin $plugin): \Illu
             'content_view' => 'plugins.missing',
             'settings' => [],
         ],
-            $manager->call(SinglePageHook::class, [], $plugin->plugin_name)[0] ?? []
+            $manager->call(SinglePageHook::class, ['user' => Auth::user()], $plugin->plugin_name)[0] ?? []
```

Comment:
Isn't this already available via type hint injection?

Reply:
Until I changed this, the `$user` in the plugin was not available. I'll be off for a week or so but I'll check that again before removing the draft status.

Reply:
You have to type hint it otherwise, it will not be injected.  Here is my test code to make sure I'm telling the truth:

```php
$object = new class {
    public function injectedUser(\Illuminate\Contracts\Auth\Authenticatable $user) {
        dump($user->toArray());
    }
    public function facadeUser() {
        dump(Auth::user()->toArray());
    }
    public function helperUser() {
        dump(auth()->user()->toArray());
    }
};

app()->call([$object, 'injectedUser'], []);
$object->facadeUser();
$object->helperUser();
```

Reply:
Not sure I get it. I am trying to feed $users in the "Page.php" page implementing PageHook in the Example Plugin.

```
class Page extends PageHook
{
    // point to the view for your plugin's settings
    // this is the default name so you can create the blade file as in this plugin
    // by omitting the variable, or point to another one

//    public string $view = 'resources.views.page';

    // The authorize method will determine if the user has access to this page.
    // if you want all users to be able to access this page simple return true
    public function authorize(\App\Models\User $user): bool
    {
        // you can check user's roles like this:
//        return $user->can('admin');

        // or use whatever you like
//        return \Carbon\Carbon::now()->dayOfWeek == Carbon::THURSDAY; // only allowed access on Thursdays!

        return true; // allow every logged in user to access
    }

```

Currently, the $user variable is not populated. This code is correcting it as far as I understand, int order to make the PluginExample work as expected.

Reply:
```php
public function authorize(\Illuminate\Contracts\Auth\Authenticatable $user): bool
    return $user->can('admin');
{
```

Will work, looks like the example is wrong.

Reply:
indeed, fixed the example, it works now as expected.